### PR TITLE
feat: support round cap for polar bars

### LINKED ISSUE #4856: [feature] Support barBorderRadius on the polar bar chart
### One-line summary [问题简述]

想给环形图进度条两端设置圆角，但是好像只有柱状图有barBorderRadius属性。

### Version & Environment [版本及环境]
+ ECharts version [ECharts 版本]:
+ Browser version [浏览器类型和版本]:
+ OS Version [操作系统类型和版本]:

### Expected behaviour [期望结果]

image

### ECharts option [ECharts配置项]

```javascript
option = {

}

```

### PR DESCRIPTION
**Pull request by Ovilia**
Fix #4856 .

image

Set `roundCap` to be `true` (default `false`) for polar bars will have this effect.

```js
var option = {
    angleAxis: {
        max: 5
    },
    radiusAxis: {
        type: 'category',
        data: ['a', 'b', 'c'],
        z: 10
    },
    polar: {
    },
    series: [{
        type: 'bar',
        data: [1, 2, 3],
        coordinateSystem: 'polar',
        name: 'A',
        roundCap: true,
        color: 'rgba(200, 0, 0, 0.5)',
        itemStyle: {
            borderColor: 'red',
            borderWidth: 1
        }
    }],
    legend: {
        show: true,
        data: ['A']
    },
    tooltip: {
        show: true
    }
};
```

It may also help to provide an array as `polar.radius` to make rings more easily. Currently, this can be done with empty data in the inner side:

image

This will be improved in future PRs. I'll see if I can make it in this milestone. Please ignore this feature for now.

TODO:
- [x] feat: support `polar.radius`
- [ ] feat: support bar background

### CODE REVIEW

**Comment by Ovilia** [comment]
TODO: another feature could be, supporting the background of the ring. Bar series also have similar requirements.

**Comment by Ovilia** [comment]
Polar radius supports radius now. b5e6ba7
image

File: src/util/graphic.js [hunk 1]
```diff
@@ -75,6 +77,10 @@ export function extendShape(opts) {
     return Path.extend(opts);
 }
 
+export function fixClipWithShadow(conditionCheck) {
+    return fixPathClipWithShadow(Path.prototype.brush, conditionCheck);
```
Comment (100pah) [inline src/util/graphic.js]: Is there a PR for the modification of `fixPathClipWithShadow` (about adding `conditionCheck`) ?
Does `sausage` will be used as a clipping path?

File: src/util/symbol.js [hunk 2]
```diff
@@ -129,6 +129,79 @@ var Pin = graphic.extendShape({
...
+
+        var lessThanCircle = clockwise
+            ? endAngle - startAngle < Math.PI * 2
+            : startAngle - endAngle < Math.PI * 2;
+
+        if (lessThanCircle) {
+            ctx.moveTo(unitStartX * r0 + x, unitStartY * r0 + y);
+
+            ctx.arc(unitStartX * rCenter + x, unitStartY * rCenter + y, dr,
+                -Math.PI + startAngle, startAngle, !clockwise);
```
Comment (100pah) [inline src/util/symbol.js]: Maybe it is not important but I still prefer to point out the code style issue about the line wrapping:

By google code style the it recommends "at least double normal indent" [1][2]. That is, if the normal is 2 space, the double should be 4 space. If the normal is 4 space, the double should be 8 space. So here should be:
```js
ctx.arc(unitStartX * rCenter + x, unitStartY * rCenter + y, dr,
        -Math.PI + startAngle, startAngle, !clockwise);
```

By ecomfe code style (where most of the echarts code style come from), the back parenthesis is recommended to placed in a separate line (align with the indent of the line where the start parenthesis is) and no need to double the indent. So the line wrap should be:
```js
ctx.arc(unitStartX * rCenter + x, unitStartY * rCenter + y, dr,
    -Math.PI + startAngle, startAngle, !clockwise
);
```

Both of the two styles are reasonable (to distinguish between the line break and deeper block of code). But the current way is not reasonable enough I think.

[1] https://google.github.io/styleguide/jsguide.html#formatting-indent
[2] https://google.github.io/styleguide/jsguide.html#formatting-function-arguments

File: src/util/graphic.js (hunk 1 above)
Comment (Ovilia) [reply src/util/graphic.js]: Sorry, I forgot to make a PR. It's available now: https://github.com/ecomfe/zrender/pull/513/files

**Comment by zifreestyle** [comment]
代码在本地显示为圆角，但在服务器上又是平的了

**Comment by maplness** [comment]
Is there an offset property that allows those bars don't need to start from 0 point for all?

**Comment by gaoqiang19514** [comment]
nice work!

**Comment by iaminvictus1993** [comment]
mark,bar stack 做环形图倒角

**Pull request merged**